### PR TITLE
Add label selector to lower memory usage

### PR DESCRIPTION
**Pull request by ericgraf**
…arge clusters

Help lower memory usage by filtering the pods by label.

Related PM: https://github.com/giantswarm/giantswarm/issues/16574

### CODE REVIEW

File: network/network.go [hunk 1]
```diff
@@ -175,7 +175,7 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 
 	var wg sync.WaitGroup
 
-	pods, err := c.k8sClient.CoreV1().Pods(c.namespace).List(ctx, metav1.ListOptions{})
+	pods, err := c.k8sClient.CoreV1().Pods(c.namespace).List(ctx, metav1.ListOptions{LabelSelector: "app=net-exporter"})
```
Comment (ericgraf) [inline network/network.go:178]: Is there a better way to do this than hard coding the app label here?

File: network/network.go [hunk 2]
```diff
@@ -307,6 +307,7 @@ func (c *Collector) calculateNeighbours(n int, ip string, addresses []string) []
 				k = k % len(addresses)
 				neighbours = append(neighbours, addresses[k])
 			}
+			break
```
Comment (ericgraf) [inline network/network.go:310]: There is only 1 `ip` in `addresses`, once it is found doesn't make sense to keep searching for loop. 

Probably next to no impact but might save a couple small cycles.

**Comment by stone-z** [comment]
Bumping the architect-orb version will fix your CI woes

File: network/network.go (hunk 2 above)
Comment (stone-z) [reply network/network.go:310]: I don't _think_ we can assume there will only be one. There should be multiple service endpoints

File: network/network.go (hunk 1 above)
Comment (stone-z) [reply network/network.go:178]: By chance, is `c.service` the app name in this case?

Comment (MarcelMue) [reply network/network.go:178]: It's usually a good practice to put this string into a `const` at the top of a file at least.

Additionally we often use a file called `key.go` to collect these kinds of "magic" strings :)

Also: usually we declare a separate variable first e.g.:
```
opts := metav1.ListOptions{
    LabelSelector: "app=net-exporter"
}
```

File: network/network.go [hunk 3]
```diff
@@ -174,8 +176,12 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
...
 
 	var wg sync.WaitGroup
-
-	pods, err := c.k8sClient.CoreV1().Pods(c.namespace).List(ctx, metav1.ListOptions{LabelSelector: "app=net-exporter"})
+	
+    opts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf( "app=%s", appName ),
+	}
+	
+	pods, err := c.k8sClient.CoreV1().Pods(c.namespace).List(ctx, opts)
```
Comment (stone-z) [inline network/network.go]: Nit: whitespace is weird here

File: helm/net-exporter/templates/daemonset.yaml [hunk 4]
```diff
@@ -94,9 +94,9 @@ spec:
           # side-effects to measurements.
           requests:
             cpu: 50m
-            memory: 75Mi
+            memory: 150Mi
```
Comment (MarcelMue) [inline helm/net-exporter/templates/daemonset.yaml:97]: Should we actually do this?

Comment (stone-z) [reply helm/net-exporter/templates/daemonset.yaml:97]: Eric and I talked about it and didn't see any harm in bumping it until we can verify that the pod list fix actually has enough impact in the busier clusters

Comment (stone-z) [reply helm/net-exporter/templates/daemonset.yaml:97]: Opened https://github.com/giantswarm/giantswarm/issues/16797 to remind us to revert it

Comment (MarcelMue) [reply helm/net-exporter/templates/daemonset.yaml:97]: How about only bumping the limits but not the requests? It seems a bit harsh for small clusters.

Comment (stone-z) [reply helm/net-exporter/templates/daemonset.yaml:97]: Yeah you're right, that's better. +1 for only bumping the limit

File: network/network.go [hunk 5]
```diff
@@ -35,6 +35,8 @@ const (
 	// Having a value of 2 means that 2 specific net-exporters need to be down
 	// for one net-exporter to not be dialed, without exposing very high cardinality metrics.
 	numNeighbours = 2
+
+	appName = "net-exporter"
```
Comment (MarcelMue) [inline network/network.go:39]: Super nit:
Can you move this up into the same block as "namespace" (Line 23)

File: helm/net-exporter/templates/daemonset.yaml (hunk 4 above)
Comment (ericgraf) [reply helm/net-exporter/templates/daemonset.yaml]: +1 Thanks for pointing that out. Made the change.

File: network/network.go (hunk 5 above)
Comment (ericgraf) [reply network/network.go]: This is exactly the feedback I'm looking for. Thanks!

File: network/network.go (hunk 3 above)
Comment (ericgraf) [reply network/network.go]: The original code uses tabs, I'm using code server which is putting spaces by default. Just to make things interesting..

**Comment by MarcelMue** [comment]
Please assign yourself and set ready for review if tested :)

File: network/network.go (hunk 3 above)
Comment (ericgraf) [reply network/network.go]: I think I got it this time :)

**Comment by ericgraf** [comment]
Tested working on ginger/nm4o9 but don't see any significant memory improvements with the changes. This is tested with ~10 nodes

Screenshot from 2021-04-19 14-55-39

**Comment by MarcelMue** [comment]
We should see on some big clusters. So fine with releasing this.

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**